lib/streamlit/source_util.py

```diff
@@ -101,10 +88,10 @@ def page_name_and_icon(script_path: Path) -> Tuple[str, str]:
     if not name:
         name = extraction.group(1)
 
-    extracted_icon = re.search(PAGE_ICON_REGEX, name)
-    if extracted_icon is not None:
-        icon = str(extracted_icon.group(1))
-        name = re.sub(PAGE_ICON_REGEX, "", name)
+    first_char = name[0]
```

Comment:
I'm not sure if this works with all emojis since there are those combined emojis (emojis are weird 😬), e.g.:

<img width="272" alt="image" src="https://user-images.githubusercontent.com/2852129/174136255-0c74c28f-1281-4705-adff-32197e80194a.png">

Have you tested it with that as well? The one from the test case is a single emoji: 

<img width="272" alt="image" src="https://user-images.githubusercontent.com/2852129/174137359-af749b47-c5ea-47c0-ad7d-b1d886fd62f2.png">

Reply:
Ah, yeah emoji are weird 😬 

I reworked this to use another function in the `emoji` package to make this work properly with emoji with length >1.